content/blogposts/2023/guest-post-arafat-khan-php-python.md

```diff
@@ -0,0 +1,254 @@
+
+
+Obviously, we are dealing with LLMs here, so nothing is perfect. Time and again, LLMs would hallucinate and give me imperfect responses, but in those cases, I would have to use my programming background and some common sense to tweak the code. I would also try my best to write as much code myself coz that teaches you the intricacies of the syntax and the common mistakes that people tend to make, and it’s generally a great learning experience.
+
+
+Cody can be surprisingly helpful even when you just want to write all the code yourself. One of the classic mistakes while writing PHP as a newbie is missing the `$` sign before variables, as shown below. With Cody, it’s easy just to ask it to fix them for you, and you are good to go. It’s much better than the natural IntelliSense fixes that come with VS Code by default. I appreciate this feature as it saves me from tedious single-character bug fixes.
+
+<Figure
+  src="https://storage.googleapis.com/sourcegraph-assets/blog/guest-post-arafat-khan-php-python/guest-post-arafat-khan-php-python-image-5-full.png"
+  caption="PHP Variable fixes with Cody"
+/>
+
+The other thing that I really enjoyed was the improved speed of coding.
+
+Typically, the coding process goes like this:
+
+1. Conceptualize
+2. Draft an imperfect code
+3. Refine
+4. Comment
+    
+However, with Cody, I experienced a much faster approach:
+
+1. Conceptualize
+2. Comment
+3. Generate a near-perfect code
+4. Refine
+
+<Figure
+  src="https://storage.googleapis.com/sourcegraph-assets/blog/guest-post-arafat-khan-php-python/image_6.jpg"
+  caption="Couldn’t resist adding an AI image to show the contrast of two Coders"
+/>
+
+Sometimes, I would just describe what the function is “supposed” to do in a comment, and just like magic, Cody would write it for me.
+
+The above example is a classic Cody’s autocomplete feature. It works surprisingly well for all kinds of code completion tasks, especially when your repo already has a reasonable amount of code to give it context. This autocomplete system is grounded in a complex foundation, and for those interested in its intricacies, you can delve deeper [here](https://about.sourcegraph.com/blog/the-lifecycle-of-a-code-ai-completion).
+
+All these features, such as_ autocomplete, explanation, code fixes, and testing,_ made my job as a programmer a lot easier. I was able to build the building blocks of my code, and then, inch by inch, I was getting closer to completing the entire project. As I was building more and more of this codebase, a question that crossed my mind was:
+
+## Why use Cody when you have GPT-4?
```

Comment:
```suggestion
## Why use Cody when you have Chat GPT-4?
```

Ideally its best to update this segment of this blog to say that now Cody will also support GPT 4.5 Turbo which does have the very high context window coz some people(Myself included) would be such a huge fan of it that we are more than happy to pay for it and even use our own keys. Folks already made the knowledge of its existence public in [this blog](https://about.sourcegraph.com/blog/cody-vscode-0-16-release) so perhaps its best to change the argument to. Instead of copy pasting in ChatGPT(which is what I used to do before) now you got it all in Cody.